librz/analysis/analysis.c

```diff
@@ -481,15 +481,72 @@ RZ_API int rz_analysis_archinfo(RzAnalysis *analysis, RzAnalysisInfoType query)
 	return value;
 }
 
-#define K_NORET_ADDR(x) sdb_fmt("addr.%" PFMT64x ".noreturn", x)
-#define K_NORET_FUNC(x) sdb_fmt("func.%s.noreturn", x)
+static bool sdb_noret_addr_set(Sdb *db, ut64 addr, bool v, ut32 cas) {
+	char *key = rz_str_newf("addr.%" PFMT64x ".noreturn", addr);
```

Comment:
Address strings can be optimized by using stack and rz_strf().The one with function name better leave dynamic to avoid artificial limitations on the length of the function name.